### PR TITLE
Fix disconnect deadlock

### PR DESCRIPTION
**Pull request by sawbe**
- Removes deadlock from .Wait()
- TaskServerConnectionCheck handles all disconnects to simplify threading

### CODE REVIEW

File: source/GeoVR.Connection/ClientConnection.cs [hunk 1]
```diff
@@ -426,45 +432,57 @@ private void DisconnectFromVoiceServer()
...
+                            logger.Error($"TaskVoiceServerTransmit not running. ({connection.Callsign})");
+                            await disconnectReason(DisconnectReasons.InternalLibraryError30);
+                            break;
+                        }
+                        stopWatch.Restart();
                     }
-                    stopWatch.Restart();
                 }
-                Thread.Sleep(500);
+                Thread.Sleep(10);
```
Comment (macaba) [inline source/GeoVR.Connection/ClientConnection.cs:483]: Short delay time, is this needed? 100Hz seems like a lot.

Comment (sawbe) [reply source/GeoVR.Connection/ClientConnection.cs:483]: No, leftover from testing. The sleep time will impact the time it takes for a user requested disconnect though, maybe up it to 200 or something? 

I’m sure there’s a better way to signal this, it was just the easiest way to have the existing task take care of all disconnect cases

**Comment by sawbe** [comment]
ManualResetEventSlim instead of the shorter sleep.

**Comment by macaba** [comment]
Nice fix.

At some point this class needs an overhaul but I'm happy with the high quality bandages applied.

**Pull request merged**